bliss/encoder/metrics.py

```diff
@@ -149,55 +149,61 @@ def _get_postfix(self):
 
 
 class DetectionPerformance(FilterMetric):
-    """Calculates detection and classification metrics between two catalogs.
-    Note that galaxy classification metrics are only computed when the values are available in
-    both catalogs.
-    """
-
     def __init__(
         self,
-        mag_band: int = 2,
-        mag_bin_cutoffs: list = None,
-        bin_unit_is_flux: bool = False,
+        bin_cutoffs: list = None,
+        ref_band: int = 2,
+        bin_type: str = "Flux",
         exclude_last_bin: bool = False,
         filter_list: List[CatFilter] = None,
     ):
         super().__init__(filter_list if filter_list else [NullFilter()])
 
-        self.mag_band = mag_band
-        self.mag_bin_cutoffs = mag_bin_cutoffs if mag_bin_cutoffs else []
-        self.bin_unit_is_flux = bin_unit_is_flux
+        self.bin_cutoffs = bin_cutoffs if bin_cutoffs else []
+        self.ref_band = ref_band
+        self.bin_type = bin_type
         self.exclude_last_bin = exclude_last_bin
 
+        assert self.bin_type in {"Flux", "Mag"}, "invalid bin type"
+
         detection_metrics = [
             "n_true_sources",
             "n_est_sources",
             "n_true_matches",
             "n_est_matches",
         ]
         for metric in detection_metrics:
-            n_bins = len(self.mag_bin_cutoffs) + 1  # fencepost
+            n_bins = len(self.bin_cutoffs) + 1  # fencepost
             init_val = torch.zeros(n_bins)
             self.add_state(metric, default=init_val, dist_reduce_fx="sum")
 
     def update(self, true_cat, est_cat, matching):
         assert isinstance(true_cat, FullCatalog), "true_cat should be FullCatalog"
         assert isinstance(est_cat, FullCatalog), "est_cat should be FullCatalog"
 
-        if self.mag_band is not None:
-            true_mags = true_cat.on_fluxes if self.bin_unit_is_flux else true_cat.magnitudes
-            true_mags = true_mags[:, :, self.mag_band].contiguous()
-            est_mags = est_cat.on_fluxes if self.bin_unit_is_flux else est_cat.magnitudes
-            est_mags = est_mags[:, :, self.mag_band].contiguous()
+        if self.ref_band is not None:
+            if self.bin_type == "Flux":
```

Comment:
It'd be nice if the catalog classes did the conversions among physical units. Based on Camille's comments yesterday, it seems like we should be using magnitude more often

Reply:
In `TileCatalog`, perhaps `on_fluxes` could be converted a method that takes units as an argument? Or we could replace it with three properties: `on_nmgy`, `on_njy`, and `on_mag`.

Reply:
Good idea. I'll do it.

Reply:
I change it to `on_nmgy`, `on_njy` and `on_mag`